Detectors/ITSMFT/MFT/calibration/src/NoiseSlotCalibrator.cxx

```diff
@@ -0,0 +1,147 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   NoiseSlotCalibrator.cxx
+
+#include "MFTCalibration/NoiseSlotCalibrator.h"
+
+#include "FairLogger.h"
+#include "TFile.h"
+#include "DataFormatsITSMFT/Digit.h"
+#include "DataFormatsITSMFT/ClusterPattern.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+
+namespace o2
+{
+using Slot = calibration::TimeSlot<o2::itsmft::NoiseMap>;
+
+namespace mft
+{
+bool NoiseSlotCalibrator::processTimeFrame(gsl::span<const o2::itsmft::Digit> const& digits,
+                                           gsl::span<const o2::itsmft::ROFRecord> const& rofs)
+{
+  calibration::TFType nTF = rofs[0].getBCData().orbit / mHBFperTF;
```

Comment:
The calibration may start from any orbit, so the nTF in general will not reflect the TFID. In other time-slot-based calibrations we are using `auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header)->startTime;`. This is also not really correct, but at least, once we come to agreement on the common TF counter definition, we can change all centrally. I would suggest to use the same: extract the TF counter in the CalibrationSpec and pass it to the Calibrator, e.g. like in `https://github.com/AliceO2Group/AliceO2/blob/c2f765d42f324beb1e1ecda9cf5ba4acf29704cd/Detectors/TOF/calibration/testWorkflow/LHCClockCalibratorSpec.h#L51-L54`
Then, you also will not need the `mHBFperTF`.
Actually, for you the specific slot should be irrelevant, since you have just 1 with 0 - infinity span.

Reply:
Thanks for pointing this out. Indeed I had some troubles with this. I think the same line is in the ITS code.